tests/environment.js

```diff
@@ -0,0 +1,27 @@
+const NodeEnvironment = require('jest-environment-node');
+const request = require('request-promise-native');
+const uuid = require('uuid/v1');
+
+class FastlyEnvironment extends NodeEnvironment {
+  async setup() {
+    await super.setup();
+
+    const domain = process.env.FASTLY_DOMAIN;
+    const baseUrl = `https://${domain}`;
+
+    // Test client.
+    this.global.http = request.defaults({
+      baseUrl,
```

Comment:
Might want to set this explicitly to a non-shield POP (and set a `Host` header), not sure where the Travis builds are happening but might be inconsistent.

Reply:
Should also probably test shield vs edge.